pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTUserEnum.java

```diff
@@ -19,7 +22,14 @@ public Object jjtAccept(ApexParserVisitor visitor, Object data) {
 
     @Override
     public String getImage() {
-        return node.getClass().getName();
```

Comment:
Can you try using

```
return node.getDefiningType().getApexName();
```

instead? We should try to avoid using reflection, since this would fail late only at runtime.
I know, that we use the same code in ASTUserClass.... we should change both places and add unit tests for the `getImage()` method.

Reply:
In the following case, it returns Foo.FooEnum. I could return the substring after the last '.' .
```
public class Foo {
    public enum FooEnum { }
}
```

Reply:
Fixed for ASTUserClass, ASTUserEnum, ASTUserInterface, and ASTUserTrigger in ff7abcf224b1ac55c1b5b8e9ef7d33c0ade4d43f. I added tests as well, except for ASTUserTrigger which already had tests.

Reply:
Awesome!